streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/utils/YarnQueueLabelExpression.java

```diff
@@ -0,0 +1,135 @@
+      Pattern.compile(String.format("^(%s)(.%s)*(%s(%s)(,%s)*)?$", REGEX, REGEX, AT, REGEX, REGEX));
+
+  public static final String QUEUE_LABEL_FORMAT = "%s" + AT + "%s";
+
+  private final String queue;
+  private @Nullable final String labelExpression;
+
+  private YarnQueueLabelExpression(String queue, String labelExpression) {
+    this.labelExpression = StringUtils.isEmpty(labelExpression) ? null : labelExpression;
+    this.queue = queue;
+  }
+
+  public Optional<String> getLabelExpression() {
+    return Optional.ofNullable(labelExpression);
+  }
+
+  public @Nonnull String getQueue() {
+    return queue;
+  }
+
+  @Override
+  public String toString() {
+    return StringUtils.isEmpty(labelExpression)
+        ? queue
+        : String.format(QUEUE_LABEL_FORMAT, queue, labelExpression);
+  }
+
+  private static boolean isValid(String queueLabel, boolean ignoreEmpty) {
+    if (StringUtils.isEmpty(queueLabel)) {
+      return ignoreEmpty;
+    }
+    return QUEUE_LABEL_PATTERN.matcher(queueLabel).matches();
+  }
+
+  // Visible for test.
+  public static boolean isValid(String queueLabel) {
+    return isValid(queueLabel, false);
+  }
+
+  /**
+   * Only check the validation of queue-labelExpression when using yarn application or yarn-session
+   * mode.
+   *
+   * @param executionMode execution mode.
+   * @param queueLabel queueLabel expression.
+   */
+  public static void checkQueueLabelIfNeed(int executionMode, String queueLabel) {
+    if ((ExecutionMode.YARN_SESSION == ExecutionMode.of(executionMode)
+            || ExecutionMode.YARN_APPLICATION == ExecutionMode.of(executionMode))
+        && !YarnQueueLabelExpression.isValid(queueLabel, true)) {
+      throw new IllegalArgumentException(ERR_HINTS);
+    }
+  }
+
+  // Visible for test.
+  public static YarnQueueLabelExpression of(@Nonnull String queueLabelExpr) {
+    if (isValid(queueLabelExpr, false)) {
+      String[] strs = queueLabelExpr.split(AT);
+      if (strs.length == 2) {
+        return new YarnQueueLabelExpression(strs[0], strs[1]);
+      }
+      return new YarnQueueLabelExpression(strs[0], null);
+    }
+    throw new IllegalArgumentException(ERR_HINTS);
+  }
+
+  public static YarnQueueLabelExpression of(
+      @Nonnull String queue, @Nullable String labelExpression) {
+    YarnQueueLabelExpression queueLabelExpression =
+        new YarnQueueLabelExpression(queue, labelExpression);
+    if (isValid(queueLabelExpression.toString(), false)) {
+      return queueLabelExpression;
+    }
+    throw new IllegalArgumentException(ERR_HINTS);
+  }
+
+  /**
+   * Add the queue and label expression into the map as the specified configuration items. For
+   * {@link ConfigConst#KEY_YARN_APP_QUEUE} & {@link ConfigConst#KEY_YARN_APP_NODE_LABEL()}.
+   *
+   * @param queueLabelExp queue and label expression
+   * @param map the target properties.
+   */
+  public static <V> void addQueueLabelExprInto(String queueLabelExp, @Nonnull Map<String, V> map) {
```

Comment:
Pass in a Map for filling. Not a good way, how about ? 

```
public static Map<String, String> getQueueLabelMap(String queueLabelExp) {
    if (StringUtils.isEmpty(queueLabelExp)) {
      return Collections.emptyMap();
    }
    YarnQueueLabelExpression yarnQueueLabelExpression = of(queueLabelExp);
    Map<String, String> map = new HashMap<>(2);
    yarnQueueLabelExpression
        .getLabelExpression()
        .ifPresent(labelExp -> map.put(ConfigConst.KEY_YARN_APP_NODE_LABEL(), labelExp));
    map.put(ConfigConst.KEY_YARN_APP_QUEUE(), yarnQueueLabelExpression.queue);
    return map;
  }

```

Also I don't know if both "KEY_YARN_APP_NODE_LABEL" and "KEY_YARN_APP_QUEUE" need to put

Reply:
If the label or queue exists, the corresponding value would be put.

Reply:
> If the label or queue exists, the corresponding value would be put.

get it